### PR TITLE
Add small es6 improvements

### PR DESCRIPTION
**Pull request by kjgalvan**
- Fix compiler warnings around types (nullability, expected vs. received)
- Enforce column limit (except helper.js line 62 which will be handled in PR #35)
- Change to const and lets
- Change to method shorthands
- Use default parameters

### CODE REVIEW

File: src/helper/helper.js [hunk 1]
```diff
@@ -236,18 +235,18 @@ function buildAbstractModelInterface_(dataLayerHelper) {
...
  * with any arguments passed in.
  *
- * @param {Array.<Object>} command The array containing the key with the
+ * @param {*} command The array containing the key with the
  *     method to execute and optional arguments for the method.
- * @param {Object|Array} model The current dataLayer model.
+ * @param {!Object<*>|!Array<*>} model The current dataLayer model.
  * @private
  */
 function processCommand_(command, model) {
```
Comment (Alex7Li) [inline src/helper/helper.js:243]: Shouldn't the first param be of !Array<*> type? I think getting to this point of the code is only possible if plain.type gives 'array', so you can safely cast it

File: src/helper/helper.js [hunk 2]
```diff
@@ -235,7 +234,7 @@ function buildAbstractModelInterface_(dataLayerHelper) {
  * If the method is a valid function of the value, the method will be applies
  * with any arguments passed in.
  *
- * @param {*} command The array containing the key with the
+ * @param {!Array<*>} command The array containing the key with the
  *     method to execute and optional arguments for the method.
  * @param {!Object<*>|!Array<*>} model The current dataLayer model.
```
Comment (Alex7Li) [inline src/helper/helper.js:239]: I think that the model is never an Array<*>, always Object<*>.

File: src/helper/helper.js [hunk 3]
```diff
@@ -263,7 +262,7 @@ function processCommand_(command, model) {
  * If a processor for the command has been registered, the processor function
  * will be invoked with any arguments passed in.
  *
- * @param {*} args The arguments object containing the command
+ * @param {!Object<*>} args The arguments object containing the command
  *     to execute and optional arguments for the processor.
  * @param {!Object<*>|!Array<*>} model The current dataLayer model.
```
Comment (Alex7Li) [inline src/helper/helper.js:267]: I think that the model is never an Array<*>, always Object<*>.

Comment (kjgalvan) [reply src/helper/helper.js]: Great catches, thank you! I've updated them now.

**Comment by Alex7Li** [comment]
LGTM

File: src/helper/helper.js [hunk 4]
```diff
@@ -153,80 +153,78 @@ DataLayerHelper.prototype['get'] = function(key) {
...
-      try {
-        update.call(this.abstractModelInterface_);
-      } catch (e) {
-        // Catch any exceptions to we don't drop subsequent updates.
-        // TODO: Add some sort of logging when this happens.
-      }
-    } else if (isPlainObject(update)) {
-      for (var key in update) {
-        merge_(expandKeyValue_(key, update[key]), this.model_);
+      // Checking executingListener here protects against multiple levels of
```
Comment (zikas) [inline src/helper/helper.js:177]: What happened to the indentation here?..

**Pull request merged**